libgadget/blackhole.c

```diff
@@ -1227,28 +1208,40 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
     }
 
     /* Dump feedback energy */
-    if(P[other].Type == 0) {
-        if(r2 < iter->feedback_kernel.HH && P[other].Mass > 0) {
-            if(I->FeedbackWeightSum > 0 && I->FeedbackEnergy > 0)
-            {
-                double u = r * iter->feedback_kernel.Hinv;
-                double wk = 1.0;
-                double mass_j;
-
-                if(HAS(blackhole_params.BlackHoleFeedbackMethod, BH_FEEDBACK_MASS)) {
-                    mass_j = P[other].Mass;
-                } else {
-                    mass_j = P[other].Hsml * P[other].Hsml * P[other].Hsml;
-                }
-                if(HAS(blackhole_params.BlackHoleFeedbackMethod, BH_FEEDBACK_SPLINE))
-                    wk = density_kernel_wk(&iter->feedback_kernel, u);
-
-                double * iBHPI = &BH_GET_PRIV(lv->tw)->Injected_BH_Energy[P[other].PI];
-                const double injected_BH = I->FeedbackEnergy * mass_j * wk / I->FeedbackWeightSum;
-                #pragma omp atomic update
-                (*iBHPI) += injected_BH;
-            }
+    if(P[other].Type == 0 &&
+        (r2 < iter->feedback_kernel.HH && P[other].Mass > 0) &&
+            (I->FeedbackWeightSum > 0 && I->FeedbackEnergy > 0))
+    {
+        double u = r * iter->feedback_kernel.Hinv;
+        double wk = 1.0;
+        double mass_j;
+
+        if(HAS(blackhole_params.BlackHoleFeedbackMethod, BH_FEEDBACK_MASS)) {
+            mass_j = P[other].Mass;
+        } else {
+            mass_j = P[other].Hsml * P[other].Hsml * P[other].Hsml;
         }
+        if(HAS(blackhole_params.BlackHoleFeedbackMethod, BH_FEEDBACK_SPLINE))
+            wk = density_kernel_wk(&iter->feedback_kernel, u);
+
+        const double injected_BH = I->FeedbackEnergy * mass_j * wk / I->FeedbackWeightSum;
+        /* Set a flag for star-forming particles:
+            * we want these to cool to the EEQOS via
+            * tcool rather than trelax.*/
+        if(sfreff_on_eeqos(&SPHP(other), BH_GET_PRIV(lv->tw)->a3inv)) {
+            /* We cannot atomically set a bitfield.
+             * This flag is never read in this thread loop, nor
+             * are other flags set here. So lack of atomicity is (I think) not a problem*/
+            //#pragma omp atomic write
```

Comment:
@rainwoodman  Can you confirm this is true?

Reply:
To the extend of my knowledge I agree with you if we only use a single bit this way, we do not need atomicity.

Reply:
Thanks